### ISSUE TITLE
Error code: Wsl/Service/CreateInstance/CreateVm/ConfigureNetworking/ERROR_FILE_NOT_FOUND upon executing WSL

### ISSUE THREAD

**Issue by Atari2** [microsoft/WSL#9840]
### Windows Version

Microsoft Windows [Version 10.0.19045.2788]

### WSL Version

1.1.3.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

Ubuntu 22.04

### Repro Steps

Execute wsl from a terminal.
I don't really have repro steps because it happened suddently. Yesterday everything was working fine.

### Expected Behavior

WSL execution starts as normal

### Actual Behavior

WSL hangs for a while and after a couple of minutes exits with 
```
❯ wsl
The system cannot find the file specified.
Error code: Wsl/Service/CreateInstance/CreateVm/ConfigureNetworking/ERROR_FILE_NOT_FOUND
```

**Comment by benhillis** [comment]
/logs

I suspect the virtual switch is in a bad state, but logs would confirm.

**Comment by Atari2** [comment]
[WslLogs-2023-03-24_22-45-05.zip](https://github.com/microsoft/WSL/files/11066742/WslLogs-2023-03-24_22-45-05.zip)
Logs here, sorry.

**Comment by Atari2** [comment]
@benhillis sorry to ping but any update on this?

**Comment by sm-msft** [comment]
Thank you for your patience. We need additional information and logs to further investigate this issue. 

If you had rebooted your machine since you last observed this issue, do you still observe any problems launching WSL? We would like the following logs collected for further investigation, whether or not the issue still exists.

Please collect your system networking related state using this script and share it with us: https://raw.githubusercontent.com/microsoft/SDN/master/Kubernetes/windows/debug/collectlogs.ps1

Running that script downloads other scripts on to your machine. Please look for "c:\k\debug\starthnstrace.cmd" to locate the next script, or you can download it from here: https://github.com/microsoft/SDN/blob/master/Kubernetes/windows/debug/starthnstrace.cmd

Please follow these steps to collect networking traces for WSL launch:
1. starthnstrace.cmd
2. _launch wsl_
3. netsh trace stop

The traces are written to c:\server.etl. Please collect this file and share it with us as well.

**Comment by Atari2** [comment]
Yes, the issue continues even after reboots (I rebooted the machine various times when I tried to diagnose the issue myself before making the issue here), if anything, it worsened.
Now, after various normal-use reboots during these past days, trying to run the `wsl` command just hangs forever without erroring out ever, it just stays there without any message on the command line or anything.
I collected the extra logs (now stopping the `wsl` command with Ctrl+C after various minutes of letting it run since it doesn't error out explicitely anymore, but it still hangs) but I would like to refrain from posting it publicly in this github thread and would like to know if there was any way to share them privately.

**Comment by Atari2** [comment]
> Hello Atari2 Have you solved this problem? I have encountered the same problem as you. Can you tell me how to solve it if it has already been solved? thank you

No, unfortunately the last message you see on this thread is the last time I heard from ms, I haven't received new instructions or help since.

**Comment by RrazzmatazZ** [comment]
After update my window system I got this problem too...

**Comment by olegueret** [comment]
I solved this by reinstalling VMware Workstation Player:

I got this problem after updating my Windows 10 from 21H2 to 22H2. Just before updating I uninstalled VMware Workstation Player 16 because I was not using it, and also manually removed its leftover network adapters "VMware Network Adapter VMnet1" and "VMware Network Adapter VMnet1".

I tried many things (rebooting computer, disable Hyper-V, reenable it, enable features Microsoft-Windows-Subsystem-Linux and VirtualMachinePlatform with disam.exe, etc) but nothing worked.

Then I realized that the error wsl gave had someting to do with networking (.../CreateVm/**ConfigureNetworking**/), so I downloaded and installed VMware Workstation Player 17 (which recreated its VMNet network adapters) and now WSL works again!

**Comment by siaynoq** [comment]
Confirmed 👍 
I had the exact same issue - removed VMWare Player and WSL2 stopped responding. Re-installing the Player "resolved" the issue immediately, however it would be great to understand the underlying problem as I'm not using that tool anymore and would like to remove eventually.

**Comment by XiaoSenLuo** [comment]
I got this error also when uninstall VMWare 17, and I found all Hyper-V Network Adapter removed also, I try remove the function "Windows sub Linux system" and enable it again, but cann't add Hyper-v network adapter automatically.

**Comment by ruoyi-qiao** [comment]
> I solved this by reinstalling VMware Workstation Player:
> 
> I got this problem after updating my Windows 10 from 21H2 to 22H2. Just before updating I uninstalled VMware Workstation Player 16 because I was not using it, and also manually removed its leftover network adapters "VMware Network Adapter VMnet1" and "VMware Network Adapter VMnet1".
> 
> I tried many things (rebooting computer, disable Hyper-V, reenable it, enable features Microsoft-Windows-Subsystem-Linux and VirtualMachinePlatform with disam.exe, etc) but nothing worked.
> 
> Then I realized that the error wsl gave had someting to do with networking (.../CreateVm/**ConfigureNetworking**/), so I downloaded and installed VMware Workstation Player 17 (which recreated its VMNet network adapters) and now WSL works again!

The same issue, thanks for sharing this fix.

**Comment by AidanWelch** [comment]
I get the same issue

**Comment by blackliner** [comment]
Confirmed, VMware Player 17. What a garbage SW, had to install it again to make it work.

**Comment by darkknight141** [comment]
Hello everyone, I'm facing the same problem. I also had VMware installed, I deleted it about a month ago and everything worked stably, until today. I have tried a large number of solutions: reinstalling VMP, reinstalling Hyper-V, reinstalling WSL, reinstalling VMware. More than one solution did not help. Help please

**Comment by darkknight141** [comment]
If you need any data from me, then I am ready to provide

**Comment by douglas541** [comment]
I get the same issue. Just dont download VMware at first guys.

**Comment by douglas541** [comment]
I had the same issue. Just dont download VMware.

**Comment by Raven-Singularity** [comment]
Experiencing the same issue where VMWare breaks everything.

I auto-updated to VMWare Workstation Pro 17.5.2, then manually upgraded to 17.6.0 from the Broadcom web site.  VMWare worked, but incredibly slowly in certain areas (like the VMWare BIOS, I could see it drawing individual ASCII characters onto the screen!).  However, VMWare caused my WSL2 to no longer load.  I then uninstalled VMWare, but my WSL2 remains dead in the water.  I have been completely unable to run WSL2 for the past week.

My Ubuntu virtual machine is always in status of Stopped.

### How does one go about reinstalling the WSL2 virtual network drivers?

They no longer exist, and reinstalling the apps does not reinstall these virtual network drivers!  Absurdism.

```
Failed to configure network (networkingMode Nat). To disable networking, set `wsl2.networkingMode=None` in C:\Users\MyUserName\.wslconfig
Error code: Wsl/Service/CreateInstance/CreateVm/ConfigureNetworking/HNS/ERROR_FILE_NOT_FOUND
```

Also, even following the instructions to disable networking with **wsl2.networkingMode=None** does not have any effect -- it simply complains that the file does not have an = sign in it, which it clearly does, on the first and only line in the file.

**How can I get my WSL2 running again?**

I don't care about VMWare, it is uninstalled and deleted.  I removed every trace of VMWare from the Registry, and ran into issues where it wouldn't let me delete the virtual network item with VMWare in it, it said I didn't have permissions.  I used several apps and running regedit as System and taking ownership and still wasn't able to delete 100% of the sub-keys, and they kept reappearing after I deleted them.

I also uninstalled and reinstalled all of the WSL-related components, and have tried a dozen different configurations of which things are installed (everything to do with Hyper-V, WSL, or virtualization).

Some of the commands I ran in an Administrator Windows PowerShell:

```
# netsh int ip reset all
# netsh winsock reset
# used the PSTools app to try to remove vmware registry entries
# .\PsExec.exe -i -d -s c:\windows\regedit.exe
# Restart-Service LxssManager
# Restart-Service -Force -Name hns
# Restart-NetAdapter -Name "vEthernet (WSL)"
# Restart-NetAdapter -Name "Ethernet"
# Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform
# dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart
# dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
# wsl
# Get-Service vmcompute | Restart-Service
# dism.exe /online /disable-feature /featurename:VirtualMachinePlatform /norestart
# Disable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V-Hypervisor
```

I also disabled and enabled those things in Windows Optional Features, rebooted a million times, nothing has any effect on WSL2 virtual networking, always non-existent.

I've spent around a dozen hours trying to get my WSL2 working again.  I haven't been able to work on my web sites in my virtual Ubuntu for the past week!  This is aggravatingly frustrating.

**Comment by Raven-Singularity** [comment]
I had a similar incident after installing then uninstalling VMware, which left my `Hyper-V Extensible Virtual Switch` entirely broken.  I was unable to load WSL2 for about 10 days due to the missing virtual network adapter.

What fixed it for me was removing this registry key which VMware left behind:

```
HKEY_LOCAL_MACHINE\SOFTWARE\Classes\CLSID\{3d09c1ca-2bcc-40b7-b9bb-3f3ec143a87b}
```

After removing this registry key manually, then clicking `Network Reset` on the Networking Control Panel page, my system came back up with fully working networking, including my WSL2 launching again!

Note: Network Reset made me wait 5 minutes before it rebooted the system.

Hope it helps other people who are stuck!

Other error codes and error messages I experienced:

```
Windows is still setting up class configuration for this device. (Code 56)
0x80071779
0x80041002
```
